velox/type/TimestampConversion.cpp

```diff
@@ -593,6 +613,28 @@ Status daysSinceEpochFromWeekDate(
   return Status::OK();
 }
 
+Expected<int64_t> daysSinceEpochFromWeekOfMonthDate(
+    int32_t year,
+    int32_t month,
+    int32_t weekOfMonth,
+    int32_t dayOfWeek) {
+  if (!isValidWeekOfMonthDate(year, month, weekOfMonth, dayOfWeek)) {
+    return folly::makeUnexpected(Status::UserError(
+        "Date out of range: {}-{}-{}-{}", year, month, weekOfMonth, dayOfWeek));
+  }
+  int64_t daysSinceEpochOfFirstDayOfMonth;
+  Status status =
+      daysSinceEpochFromDate(year, month, 1, daysSinceEpochOfFirstDayOfMonth);
```

Comment:
For calculating `dayOfMonth` needs to know the day of week for the first day of month. It's tricky to calculate it without using `daysSinceEpochFromDate`.  And we already do validate in `daysSinceEpochFromWeekOfMonthDate`, do we need to validate date again in `daysSinceEpochFromDate`?

Reply:
I was wondering whether we can remove the `isValidWeekOfMonthDate` since `daysSinceEpochFromDate` checks for a valid date. I'm not sure how we treat the difference between the two checks.